slatedb/src/wal_buffer.rs

```diff
@@ -57,6 +57,8 @@ struct WalBufferManagerInner {
     /// When the current WAL is ready to be flushed, it'll be moved to the `immutable_wals`.
     /// The flusher will try flush all the immutable wals to remote storage.
     immutable_wals: VecDeque<(u64, Arc<KVTable>)>,
+    /// The fatal error that occurred during the flush operation.
+    fatal: Option<SlateDBError>,
```

Comment:
Rather than this and `quit_once`, can you use the same pattern we have for Db, DbReader, etc? They have a single `WatchableOnceCell<SlateDBError>`. This would also make your code compatible with `map_slatedb_err` from #628.

Reply:
Since it's running inside DB (as opposed to DbReader), I wonder if we can simply use the DB's `WatchableOnceCell<SlateDBError>` directly?